File: examples/dict_assign.py
def test_dict_assignment():
    """Test dictionary subscript assignment"""
    # Simple assignment
    d = {}
    d["key"] = "value"
    d[42] = "number key"
    
    # Nested assignment (not yet supported in full)
    nested = {}
    nested["outer"] = {}
    # nested["outer"]["inner"] = "value"  # TODO: This requires chained assignment
    
    return d

File: examples/dict_assign.rs
use serde_json;
use std::collections::HashMap;
#[doc = "Test dictionary subscript assignment"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_dict_assignment() -> HashMap<serde_json::Value, serde_json::Value> {
    let mut d = {
        let mut map = std::collections::HashMap::new();
        map
    };
    d.insert("key".to_string().to_string(), serde_json::json!("value"));
    d.insert(42, serde_json::json!("number key"));
    let mut nested = {
        let mut map = std::collections::HashMap::new();
        map
    };
    nested.insert(
        "outer".to_string().to_string(),
        serde_json::to_value({
            let mut map = std::collections::HashMap::new();
            map
        })
        .unwrap(),
    );
    d
}
